proposals/lip-0048.md

```diff
@@ -25,15 +25,34 @@ This LIP defines the fee system in a modular way, as currently [used in the Lisk
+
 ### Fee Token ID
 
-Each chain can configure the token used to pay fees. On the Lisk mainchain, the token used for transaction fees is the LSK token.
+Each chain can configure the token used to pay fees, as defined by `TOKEN_ID_FEE`. On the Lisk mainchain, the token used for transaction fees is the LSK token.
 
 ### Minimum Fee per Transaction
 
-As introduced in [LIP 0013][lip-0013], all transactions must have a fee greater or equal to a minimum fee (which can be zero). The minimum fee is computed from the transaction size. Chains can configure their minimum fee per byte, with a possibility to define a period of `MAX_BLOCK_HEIGHT_ZERO_FEE_PER_BYTE` blocks after the genesis block when the minimum fee per byte is set to zero.
+As introduced in [LIP 0013][lip-0013], all transactions must have a fee greater or equal to a minimum fee, which can be set to zero. For a transaction `trs`, the minimum fee is determined by its size and the proportionality constant `MIN_FEE_PER_BYTE`: 
+
+```python
+trsSize = length(encodeTransaction(trs))
+minFee = MIN_FEE_PER_BYTE * trsSize
+```
+
+Note that chains can freely configure their `MIN_FEE_PER_BYTE`, with a possibility to define a period of `MAX_BLOCK_HEIGHT_ZERO_FEE_PER_BYTE` blocks after the genesis block when it is set to zero.
+
+### Fee Pool
+
+Chains can choose to either burn the minimum fee - to discourage validators from sending zero-cost transactions in the blocks they generate - or, on the other hand, to transfer it to a dedicated pool address, `ADDRESS_FEE_POOL`. 
+
+The latter choice could be beneficial for sidechains, as fees collected in this way can be used as an incentive for validators and other protocol participants for various purposes like funding a community pool for on-chain governance spending, sharing the funds among validators proportionally per round, incentivizing ecosystem developers, supporting chain maintenance and improvement, rewarding user staking/voting activity, etc.
+
+In the Lisk mainchain, the minimum fee is always burned.
+
+### Cross-chain Message Fees
 
-Chains can choose that this minimum fee gets either burned (to avoid that validators can send transactions in the blocks they generate without cost) or transferred to the specified pool address `ADDRESS_FEE_POOL`. The latter choice could be beneficial for sidechains, as fees collected in this way can be used as an incentive for validators and other protocol participants. In the Lisk mainchain, the minimum fee is always burned.
+Cross-chain messages or CCMs facilitate interoperability by enabling cross-chain transfer of data. They require an associated fee, distinct from the usual transaction fee, paid by the relayer. Similarly to transaction fees, the minimum CCM fee can be either burned or transferred to a fee pool, depending on the chain configuration.
 
 ## Specification
 
```

Comment:
Btw I also have a comment about the verify transaction function (lines 240-255):

In line 245 we set `MIN_FEE_PER_BYTE` to 0. While I understand the intention, strictly speaking this is wrong, since we over-write the initial value.

Instead I think that what we really want to do is:

```
    if h < MAX_BLOCK_HEIGHT_ZERO_FEE_PER_BYTE:
	    # Set minFee to 0 for the first MAX_BLOCK_HEIGHT_ZERO_FEE_PER_BYTE blocks.
        minFee = 0
    else:
        trsSize = length(encodeTransaction(trs))
        minFee = MIN_FEE_PER_BYTE * trsSize
```

Reply:
Yes, I believe you're right. Do you think the implementation will change (I believe not)?